### PR TITLE
Mulitple CRON

### PR DESCRIPTION
**Pull request by crashcoq**
add function multiple cron

### CODE REVIEW

File: src/models/config.py [hunk 1]
```diff
@@ -144,7 +147,10 @@ class Config():
 
     item_ids: list
     sleep_time: int
-    schedule_cron: str
+    schedule_cron1: str
+    schedule_cron2: str
+    schedule_cron3: str
+    schedule_cron4: str
```
Comment (Der-Henning) [inline src/models/config.py:153]:
```suggestion
    schedule_cron: list[Cron]
```

File: src/models/config.py [hunk 2]
```diff
@@ -18,7 +18,10 @@
 DEFAULT_CONFIG = {
     'item_ids': [],
     'sleep_time': 60,
-    'schedule_cron': Cron('* * * * *'),
+    'schedule_cron1': Cron('* * * * *'),
+    'schedule_cron4': Cron('0 0 1 1 0'),
+    'schedule_cron4': Cron('0 0 1 1 0'),
+    'schedule_cron4': Cron('0 0 1 1 0'),
```
Comment (Der-Henning) [inline src/models/config.py:24]:
```suggestion
    'schedule_cron': [Cron('* * * * *')],
```

File: src/scanner.py [hunk 3]
```diff
@@ -171,10 +174,40 @@ def run(self) -> NoReturn:
...
-        if self.cron.cron != "* * * * *":
-            log.info("Active on schedule: %s", self.cron.description)
+        if self.cron1.cron != "* * * * *":
+            log.info("Active on schedule: %s", self.cron1.description)
+        if self.cron2.cron != "0 0 1 1 *":
+            log.info("Active on schedule: %s", self.cron2.description)
+        if self.cron3.cron != "0 0 1 1 *":
+            log.info("Active on schedule: %s", self.cron3.description)
+        if self.cron4.cron != "0 0 1 1 *":
+            log.info("Active on schedule: %s", self.cron4.description)
```
Comment (Der-Henning) [inline src/scanner.py:184]:
```suggestion
        log.info("Active on schedule: %s",
                 ", ".join([cron.description for cron in self.cron if cron.cron != "* * * * *"])
```

File: src/scanner.py [hunk 4]
```diff
@@ -171,10 +174,40 @@ def run(self) -> NoReturn:
...
+                    log.error("Job Error! - %s", sys.exc_info())
+            elif self.cron3.is_now:
+                if not running:
+                    log.info("Scanner reenabled by cron schedule.")
+                    running = True
+                try:
+                    self._job()
+                except Exception:
+                    log.error("Job Error! - %s", sys.exc_info())
+            elif self.cron4.is_now:
```
Comment (Der-Henning) [inline src/scanner.py:210]:
```suggestion
            if any([cron.isnow for cron in self.cron]):
```

File: src/config.sample.ini [hunk 5]
```diff
@@ -11,7 +11,15 @@ SleepTime = 60
...
-ScheduleCron = * * * * *
+## example:
+## ScheduleCron1 = * 18-21 * * 1-5
+## ScheduleCron2 = * 07-21 * * 6-6
+## ScheduleCron3 = * 07-12 * * 7-7
+## ScheduleCron4 = 0 0 1 1 0
+ScheduleCron1 = * * * * *
+ScheduleCron2 = 0 0 1 1 0
+ScheduleCron3 = 0 0 1 1 0
+ScheduleCron4 = 0 0 1 1 0
```
Comment (Der-Henning) [inline src/config.sample.ini:22]:
```suggestion
## Multiple cron example:
## ScheduleCron = * 18-21 * * 1-5, * 7-21 * * 6
ScheduleCron = * * * * *
```

**Review by Der-Henning — COMMENTED** [review]
Hi @crashcoq.
Thank you for your contribution.
Please make sure, that you base your PR on the current dev branch.

I would suggest we allow the possibility to add multiple crons by providing them as comma separated list. I made some comments to show what I mean.
This way the current behavior stays unchanged and you can add as many cron as you like.
The methods `Config._ini_get_cron` and `Config._env_get_cron` need to be adapted analog to the `_get_array` methods.

**Comment by crashcoq** [comment]
hello, this modification will be integrated into your github project? Cordially

**Comment by Der-Henning** [comment]
Yes, I would like to integrate your idea. But I would like to make some small changes. Would you mind if I make these changes on your branch before merging?

**Comment by crashcoq** [comment]
No worries, on the contrary, I am not an expert in python

**Comment by crashcoq** [comment]
No worries, on the contrary, I am not an expert in python

**Pull request merged**